apps/finance/app/src/components/NewTransfer/Deposit.js

```diff
@@ -159,26 +160,31 @@ class Deposit extends React.Component {
       const symbolFallback =
         tokenDataFallback(address, 'symbol', network.type) || ''
 
-      combineLatest(token.decimals(), token.symbol())
-        .first()
-        .subscribe(
-          ([decimals = decimalsFallback, symbol = symbolFallback]) =>
-            resolve({
-              symbol,
-              userBalance,
-              decimals: parseInt(decimals, 10),
-              loading: false,
-            }),
-          () => {
-            // Decimals and symbols are optional
-            resolve({
-              userBalance,
-              decimals: parseInt(decimalsFallback, 10),
-              loading: false,
-              symbol: symbolFallback,
-            })
-          }
-        )
+      const tokenData = {
+        symbol: symbolFallback,
+        userBalance,
+        decimals: parseInt(decimalsFallback, 10),
+        loading: false,
+      }
+
+      const [tokenSymbol, tokenDecimals] = await Promise.all([
+        getTokenSymbol(app, address),
+        token
+          .decimals()
+          .first()
+          .toPromise(),
+      ])
+
+      // no token symbol or decimals: resolve the fallback data
+      if (!tokenSymbol || !tokenDecimals) {
```

Comment:
I think we should just overwrite `tokenData`'s fields here, if either of the two exist:

```js
if (tokenSymbol) {
  tokenData.symbol = tokenSymbol
}
if (tokenDecimals) {
  tokenData.decimals = parseInt(tokenDecimals, 10)
}
```

Otherwise if we get one but not the other, we'll end up overwriting the other (from its fallback) by accident.